src/Mod/Arch/ArchWall.py

```diff
@@ -780,12 +970,43 @@ def execute(self,obj):
         obj.Area = obj.Length.Value * obj.Height.Value
 
     def onBeforeChange(self,obj,prop):
+        """Method called before the object has a property changed. 
+
+        Specifically, this method is called before the value changes.
+
+        If "Length" has changed, it records the old length so that .onChanged()
+        can be sure that the base needs to be changed.
+
+        Parameters
+        ----------
+        prop: string
+            The name of the property that has changed.
+        """
+
         if prop == "Length":
             self.oldLength = obj.Length.Value
 
     def onChanged(self, obj, prop):
+        """Method called when the object has a property changed.
+
+        If length has changed, will extend the length of the Base object, if
+        the Base object only has a single edge to extend.
+
+        Also hides subobjects.
+
+        Also calls ArchComponent.Component.onChanged().
+
+        Parameters
+        ----------
+        prop: string
+            The name of the property that has changed.
+        """
+
         if prop == "Length":
-            if obj.Base and obj.Length.Value and hasattr(self,"oldLength") and (self.oldLength != None) and (self.oldLength != obj.Length.Value):
+            if (obj.Base and obj.Length.Value 
+                    and hasattr(self,"oldLength") and (self.oldLength != None) 
```

Comment:
Comparison with `None` should use the `is` operator.
```py
self.oldLength is not None
```

Reply:
Resolved with 1ee0fde91b8d056ebadc7ac84ff1e639e7e6c906